HPXMLtoOpenStudio/resources/airflow.rb

```diff
@@ -1359,6 +1473,127 @@ def self.apply_infiltration_and_ventilation_fans(model, weather, vent_mech, vent
     infil_program = OpenStudio::Model::EnergyManagementSystemProgram.new(model)
     infil_program.setName(Constants.ObjectNameInfiltration + ' program')
 
+    infil_program = apply_infiltration_to_living(living_ach50, living_const_ach, infil_program, weather, has_flue_chimney)
+
+    infil_program.addLine('Set QWHV_ervhrv = 0.0')
+    if not vent_mech_erv_hrv.empty?
+      # Actuators for ERV/HRV
+      sens_name = "#{Constants.ObjectNameERVHRV} sensible load"
+      erv_sens_load_var = OpenStudio::Model::EnergyManagementSystemGlobalVariable.new(model, sens_name.gsub(' ', '_'))
+      erv_sens_load_actuator = create_sens_lat_load_actuator_and_equipment(model, sens_name, @living_space, 0.0, 0.0)
+      lat_name = "#{Constants.ObjectNameERVHRV} latent load"
+      erv_lat_load_var = OpenStudio::Model::EnergyManagementSystemGlobalVariable.new(model, lat_name.gsub(' ', '_'))
+      erv_lat_load_actuator = create_sens_lat_load_actuator_and_equipment(model, lat_name, @living_space, 1.0, 0.0)
+
+      infil_program.addLine("Set #{erv_sens_load_actuator.name} = 0.0")
+      infil_program.addLine("Set #{erv_lat_load_actuator.name} = 0.0")
+
+      # ERV/HRV/Balanced EMS load model
+      # E+ ERV model is using standard density for MFR calculation, caused discrepancy with other system types.
+      # E+ ERV model also does not meet setpoint perfectly.
+      # Therefore ERV is modeled within EMS infiltration program
+      # Air property at inlet nodes in two sides of ERV
+
+      # Common part across multiple erv/hrv programs, create only once
+      infil_program.addLine("Set ERVSupInPb = #{@pbar_sensor.name}") # oa barometric pressure
+      infil_program.addLine("Set ERVSupInTemp = #{@tout_sensor.name}") # oa db temperature
+      infil_program.addLine("Set ERVSupInW = #{@wout_sensor.name}")   # oa humidity ratio
+      infil_program.addLine('Set ERVSupRho = (@RhoAirFnPbTdbW ERVSupInPb ERVSupInTemp ERVSupInW)')
+      infil_program.addLine('Set ERVSupCp = (@CpAirFnW ERVSupInW)')
+      infil_program.addLine('Set ERVSupInEnth = (@HFnTdbW ERVSupInTemp ERVSupInW)')
+
+      infil_program.addLine("Set ERVSecInTemp = #{@tin_sensor.name}") # zone air temperature
+      infil_program.addLine("Set ERVSecInW = #{@win_sensor.name}") # zone air humidity ratio
+      infil_program.addLine('Set ERVSecCp = (@CpAirFnW ERVSecInW)')
+      infil_program.addLine('Set ERVSecInEnth = (@HFnTdbW ERVSecInTemp ERVSecInW)')
+    end
+
+    # Calculate cfm weighted average effectiveness for hvac sizing
+    # Please review this, and its integration with hvac_sizing
+    weighted_vent_mech_tot_eff = 0.0
+    weighted_vent_mech_lat_eff = 0.0
+    weighted_vent_mech_apparent_sens_eff = 0.0
+
+    # Apply balanced mechanical ventilation
```

Comment:
It looks like this is actually only applying ERV/HRV load, not handling balanced (w/o recovery) systems, right?

Reply:
Right, balanced without recovery is applied later in the airflow calculation(where only erv/hrv airlow is subtracted).

Reply:
Please update the comment then.

Reply:
Ah got it. Will do.